### PR TITLE
Create rules to load external dependencies

### LINKED ISSUE #49: WORKSPACE file should have a workspace() clause at the top
This is a recommended best practice at https://docs.bazel.build/versions/master/skylark/deploying.html#workspace

### PR DESCRIPTION
**Pull request by tengpengli**
**Commit Message:**
Use repositories.bzl to load the dependencies and
dependency_import.bzl to load the dependencies of the dependencies
**Additional Description:**
The user can use 
```
load("@bazel-rules-fuzzing//:repositories.bzl", "fuzzing_rule_dependencies")
fuzzing_rule_dependencies()
load("@bazel-rules-fuzzing//:dependency_import.bzl", "fuzzing_dependency_imports")
fuzzing_dependency_imports()
```
to complete our dependencies importing.
**Testing:**
Local testing and CI testing
**Docs Changes:** N/A
**Fix #49**

Signed-off-by: tengpeng <tengpeng.li2020@gmail.com>

	modified:   WORKSPACE
	new file:   dependency_import.bzl
	new file:   repositories.bzl

### CODE REVIEW

File: repositories.bzl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Contains the external dependencies
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def fuzzing_rule_dependencies():
```
Comment (asraa) [inline repositories.bzl:20]: Isn't rules_cc also used? (How does it get added without the workspace?)

Comment (tengpengli) [reply repositories.bzl:20]: From the `rules_cc` [readme](https://github.com/bazelbuild/rules_cc#getting-started), I should add it, but the interesting thing is that the project runs without loading `rules_cc` so long. Let me add that part!

Comment (tengpengli) [reply repositories.bzl:20]: > There is no need to use rules from this repository just yet. If you want to use rules_cc anyway, add the following to your WORKSPACE file
http_archive(
    name = "rules_cc",
    urls = ["https://github.com/bazelbuild/rules_cc/archive/TODO"],
    sha256 = "TODO",
)

The url listed above is a `404` link. I think the `rules_cc` is shipped together with the bazel, thus no need to explicitly `http_archive` it?

**Comment by tengpengli** [comment]
As https://docs.bazel.build/versions/0.22.0/external.html#transitive-dependencies pointed out, the user is responsible for loading our dependencies. In this case, I will close this PR and open a new PR to solve #49 .

File: repositories.bzl (hunk 1 above)
Comment (tengpengli) [reply repositories.bzl]: Added the `cc_rules` codes given in the repository, got `404` error returned by `http_archive`. Seems we can use the `rules_cc` directly for now.

**Comment by tengpengli** [comment]
> As https://docs.bazel.build/versions/0.22.0/external.html#transitive-dependencies pointed out, the user is responsible for loading our dependencies. In this case, I will close this PR and open a new PR to solve #49 .

 In https://docs.bazel.build/versions/master/skylark/deploying.html#dependencies, it recommends to use a macro to declare the dependencies. 

In this PR, I created a macro in `repositories.bzl` to download the dependencies, and the loading transitive dependencies is done in `WORKSPACE`.

File: WORKSPACE [hunk 2]
```diff
@@ -13,20 +13,13 @@
...
-http_archive(
-    name = "rules_python",
-    url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.2/rules_python-0.0.2.tar.gz",
-    strip_prefix = "rules_python-0.0.2",
-    sha256 = "b5668cde8bb6e3515057ef465a35ad712214962f0b3a314e551204266c7be90c",
-)
+workspace(name = "bazel_rules_fuzzing")
+
+load(":repositories.bzl", "bazel_rules_fuzzing_dependencies")
+bazel_rules_fuzzing_dependencies()
```
Comment (stefanbucur) [inline WORKSPACE:19]: According to https://docs.bazel.build/versions/master/skylark/deploying.html#dependencies, this should be `rules_fuzzing_dependencies` (drop the `bazel_` prefix).

File: WORKSPACE [hunk 3]
```diff
@@ -13,20 +13,13 @@
...
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-http_archive(
-    name = "rules_python",
-    url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.2/rules_python-0.0.2.tar.gz",
-    strip_prefix = "rules_python-0.0.2",
-    sha256 = "b5668cde8bb6e3515057ef465a35ad712214962f0b3a314e551204266c7be90c",
-)
+workspace(name = "bazel_rules_fuzzing")
+
+load(":repositories.bzl", "bazel_rules_fuzzing_dependencies")
```
Comment (stefanbucur) [inline WORKSPACE:18]: Shouldn't this be under `//fuzzing:repositories.bzl`? This is what I'm reading from the docs: https://docs.bazel.build/versions/master/skylark/deploying.html#dependencies

File: WORKSPACE [hunk 4]
```diff
@@ -13,20 +13,13 @@
...
 # limitations under the License.
 
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-http_archive(
-    name = "rules_python",
-    url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.2/rules_python-0.0.2.tar.gz",
-    strip_prefix = "rules_python-0.0.2",
-    sha256 = "b5668cde8bb6e3515057ef465a35ad712214962f0b3a314e551204266c7be90c",
-)
+workspace(name = "bazel_rules_fuzzing")
```
Comment (stefanbucur) [inline WORKSPACE:16]: From the docs, this should be "rules_fuzzing": https://docs.bazel.build/versions/master/skylark/deploying.html#workspace

Comment (tengpengli) [reply WORKSPACE:16]: For the first time I know we belong to bazelbuild 😆

File: WORKSPACE (hunk 3 above)
Comment (tengpengli) [reply WORKSPACE:18]: Oops! I should ask you the question: What is `LANG` in `<LANG>/repositories.bz`? 

Now I got the answer!

**Pull request merged**